libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/RouteProcessorRunnableTest.java

```diff
@@ -0,0 +1,133 @@
+import static org.mockito.Mockito.when;
+
+@RunWith(RobolectricTestRunner.class)
+public class RouteProcessorRunnableTest {
+
+  @Test
+  public void onRun_buildNewRouteProgressReceivesStatusAndRoute() {
+    NavigationRouteProcessor processor = mock(NavigationRouteProcessor.class);
+    NavigationStatus status = buildMockStatus();
+    DirectionsRoute route = mock(DirectionsRoute.class);
+    RouteProcessorRunnable runnable = buildRouteProcessorRunnableWith(processor, status, route);
+    runnable.updateRawLocation(mock(Location.class));
+
+    runnable.run();
+
+    verify(processor).buildNewRouteProgress(status, route);
+  }
+
+  @Test
+  public void onRun_offRouteDetectorReceivesStatus() {
+    OffRouteDetector detector = mock(OffRouteDetector.class);
+    NavigationEngineFactory factory = buildMockFactory(detector);
+    when(factory.retrieveOffRouteEngine()).thenReturn(detector);
+    NavigationStatus status = buildMockStatus();
+    RouteProcessorRunnable runnable = buildRouteProcessorRunnableWith(factory, status);
+    runnable.updateRawLocation(mock(Location.class));
+
+    runnable.run();
+
+    verify(detector).isUserOffRouteWith(status);
+  }
+
+  @Test
+  public void onRun_snapToRouteReceivesStatus() {
+    SnapToRoute snapToRoute = mock(SnapToRoute.class);
+    NavigationEngineFactory factory = buildMockFactory(snapToRoute);
+    NavigationStatus status = buildMockStatus();
+    RouteProcessorRunnable runnable = buildRouteProcessorRunnableWith(factory, status);
+    Location rawLocation = mock(Location.class);
+    runnable.updateRawLocation(rawLocation);
+
+    runnable.run();
+
+    verify(snapToRoute).getSnappedLocationWith(status, rawLocation);
+  }
+
+  private RouteProcessorRunnable buildRouteProcessorRunnableWith(NavigationRouteProcessor processor,
+                                                                 NavigationStatus status, DirectionsRoute route) {
+    MapboxNavigationOptions options  = MapboxNavigationOptions.builder().build();
+    MapboxNavigator navigator = mock(MapboxNavigator.class);
+    when(navigator.retrieveStatus(any(Date.class), any(Long.class))).thenReturn(status);
+    MapboxNavigation navigation = mock(MapboxNavigation.class);
+    when(navigation.options()).thenReturn(options);
+    when(navigation.getRoute()).thenReturn(route);
+    when(navigation.retrieveMapboxNavigator()).thenReturn(navigator);
+    when(navigation.retrieveEngineFactory()).thenReturn(new NavigationEngineFactory());
+    return new RouteProcessorRunnable(
+      processor,
+      navigation,
+      mock(Handler.class),
+      mock(Handler.class),
+      mock(RouteProcessorBackgroundThread.Listener.class)
+    );
+  }
+
+  private RouteProcessorRunnable buildRouteProcessorRunnableWith(NavigationEngineFactory factory,
+                                                                 NavigationStatus status) {
+    MapboxNavigationOptions options  = MapboxNavigationOptions.builder().build();
```

Comment:
Are you opposed to using fakers to create non-mocked objects like `MapboxNavigationOptions`? Like this for example: https://github.com/mapbox/mapbox-navigation-android/blob/master/libandroid-navigation-ui/src/test/java/com/mapbox/services/android/navigation/ui/v5/instruction/BannerComponentsFaker.java

Reply:
@devotaaabel would this be to be more explicit regarding naming within the test?

Reply:
@danesfeder It's mostly just to remove the clutter from the actual test classes.